src/Build/Logging/SimpleErrorLogger.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+#nullable disable
+
+namespace Microsoft.Build.Logging.SimpleErrorLogger
+{
+    public class SimpleErrorLogger : INodeLogger
+    {
+        public StringBuilder errorList;
```

Comment:
Why keep a single `StringBuilder` instead of an array of errors (or an array of formatted strings)?

Reply:
I don't think it matters too much in this case, to be honest. This isn't intended to be a particularly clever or detailed logger. It isn't harder to append to a string builder or a list, and when accessing the errors at the other end, I'd just concatenate them with \n in between anyway, so I may as well do that here. If you foresee other uses for this logger (I don't) and want it to be a list, I can make that happen.

Reply:
> when accessing the errors at the other end, I'd just concatenate them with \n in between anyway

I think the mismatch is I expected see an array of them in the JSON in the failure case.

Reply:
That was one of the options proposed, but that question stayed open for a while with no input provided. The design is not changing on this point now.

Reply:
I thought we got to a point where errors were able to be emitted to stderr?

Reply:
That is what I remember too, now that you mention it.

Reply:
They are; this is part of that mechanism.

This logger is enabled in place of the standard console logger when we're attaching loggers as part of processing command line switches. It only listens for errors and keeps track of them as we go along. If we ultimately decide that we want to show errors, they're all emitted to stderr in the end.

I could have emitted them to stderr as we go, and I think I had a reason for why I did it this way instead, but it's been a while, and I don't remember.

Reply:
My attempts did not show them logged to stderr--see the start of this thread.

Reply:
As I said in that thread, I see the "error" condition as more narrow than you do. My vision for this feature is that it does not log errors unless it is unable to do what it tries to do. If you use the error task, that is a legitimate "error" but not a catastrophic one that prevents us from telling you the result of the target (failure).

That means that the two cases in which I log to stderr are:
1. We are only evaluating your project rather than running a full build, and we get an InvalidProjectFileException when trying to load your project.
2. We are executing a full build, but we fail to save off the built project, presumably because of a catastrophic failure during build.

In all other cases, the design I'm following is to just provide all the information we can without erroring or writing to stderr.